addons/udes_stock_refactoring/models/stock_move.py

```diff
@@ -359,6 +359,132 @@ def _refactor_action_by_maximum_quantity(self, maximum_qty):
 
         return self | new_pickings.move_lines
 
+    def _refactor_action_by_maximum_weight(self, maximum_weight):
+        """
+        Split move_lines out into pickings with a maximum weight
+        This first tries to create pickings with a single move with the maximum allowed
+        Then combines the remaining moves into pickings with maximum allowed
+        """
+        StockMove = self.env["stock.move"]
+
+        if maximum_weight < 1:
+            raise ValidationError(_("Cannot split quants into weight: %i") % maximum_weight)
+
+        all_moves = StockMove.browse()
+
+        for picking, moves in self.groupby("picking_id"):
+            moves_to_refactor = []
+            grouped_moves = moves._split_and_group_by_weight(maximum_weight)
+            max_range = len(grouped_moves)
+            # Build iterable of grouped moves to refactor, which can be passed into refactor_by_move_groups.
+            # Give each set of moves a unique group name so the pickings are kept separate.
+            for idx, moves_to_keep in enumerate(grouped_moves[:max_range]):
+                group_name = f"{picking.name}-{idx:0>3}"
+                moves_to_refactor.insert(0, (group_name, moves_to_keep))
```

Comment:
Question: Why do we insert at index 0 moves_to_keep?

Reply:
No reason, can probably just append as order doesn't matter